packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -59,7 +59,9 @@ export function detectPreviewType(document: DiscoveryDocument, file?: DocumentFi
   const isJsonType = isJsonFile(document);
   const isCsvType = isCsvFile(document);
   if (document.html && !isJsonType && !isCsvType) {
-    return 'HTML';
+    if (hasTextMappings || !hasPassage) {
```

Comment:
Please add a comment here about _why_ this code is like this, since it is not obvious just from the code.